gax-java/gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -414,6 +426,14 @@ public B setClock(ApiClock clock) {
       return self();
     }
 
+    // This meant to be used internally by GAPIC clients to configure an endpoint.
+    // Users should avoid settings this as it could change their endpoint.
+    @InternalApi
+    public B setHostServiceName(String hostServiceName) {
```

Comment:
Do we have to expose a setter for it? Since the host service name is a fixed String, I think we should implement it in a similar way as the to-be-deprecated default endpoint, that we generate a getter that returns the default host service name.

Reply:
I see, I think that makes sense. I'll update it.